[CI] ProfileServiceTests testActivateWhenShouldSkipUpdateForActivateReturnsFalseFirst failing

**Build scan:**
https://gradle-enterprise.elastic.co/s/54lavnek2dx6y/tests/:x-pack:plugin:security:test/org.elasticsearch.xpack.security.profile.ProfileServiceTests/testActivateWhenShouldSkipUpdateForActivateReturnsFalseFirst

**Reproduction line:**
`./gradlew ':x-pack:plugin:security:test' --tests "org.elasticsearch.xpack.security.profile.ProfileServiceTests.testActivateWhenShouldSkipUpdateForActivateReturnsFalseFirst" -Dtests.seed=42D2AF16C52CCE17 -Dtests.locale=de-AT -Dtests.timezone=EST5EDT -Druntime.java=17`

**Applicable branches:**
main

**Reproduces locally?:**
Yes

**Failure history:**
https://gradle-enterprise.elastic.co/scans/tests?tests.container=org.elasticsearch.xpack.security.profile.ProfileServiceTests&tests.test=testActivateWhenShouldSkipUpdateForActivateReturnsFalseFirst

**Failure excerpt:**
```
java.lang.AssertionError: 
Expected: sameInstance(<org.elasticsearch.transport.RemoteTransportException: []>)
     but: was <org.elasticsearch.index.engine.VersionConflictEngineException: : version conflict,>

  at __randomizedtesting.SeedInfo.seed([42D2AF16C52CCE17:7DECD8DED393D787]:0)
  at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
  at org.junit.Assert.assertThat(Assert.java:956)
  at org.junit.Assert.assertThat(Assert.java:923)
  at org.elasticsearch.xpack.security.profile.ProfileServiceTests.testActivateWhenShouldSkipUpdateForActivateReturnsFalseFirst(ProfileServiceTests.java:965)
  at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(NativeMethodAccessorImpl.java:-2)
  at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
  at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
  at java.lang.reflect.Method.invoke(Method.java:568)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1758)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:946)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:982)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:996)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:44)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:45)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:843)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:490)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:955)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:840)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:891)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:902)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:38)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:47)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.lambda$forkTimeoutingTask$0(ThreadLeakControl.java:850)
  at java.lang.Thread.run(Thread.java:833)

```

--- a/x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/profile/ProfileServiceTests.java
+++ b/x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/profile/ProfileServiceTests.java
@@ -908,7 +908,6 @@ public class ProfileServiceTests extends ESTestCase {
         verify(service, never()).doUpdate(any(), anyActionListener());
     }
 
-    @AwaitsFix(bugUrl = "https://github.com/elastic/elasticsearch/issues/89764")
     public void testActivateWhenShouldSkipUpdateForActivateReturnsFalseFirst() throws IOException {
         final ProfileService service = spy(profileService);
         doAnswer(
@@ -922,10 +921,11 @@ public class ProfileServiceTests extends ESTestCase {
 
         // Throw version conflict on update to force GET document
         final Exception updateException;
+        final VersionConflictEngineException versionConflictException = new VersionConflictEngineException(mock(ShardId.class), "", "");
         if (randomBoolean()) {
-            updateException = new RemoteTransportException("", new VersionConflictEngineException(mock(ShardId.class), "", ""));
+            updateException = new RemoteTransportException("", versionConflictException);
         } else {
-            updateException = new VersionConflictEngineException(mock(ShardId.class), "", "");
+            updateException = versionConflictException;
         }
         doAnswer(invocation -> {
             @SuppressWarnings("unchecked")
@@ -963,7 +963,8 @@ public class ProfileServiceTests extends ESTestCase {
         if (secondCheckResult) {
             assertThat(future.actionGet(), equalTo(versionedDocument.toProfile(Set.of())));
         } else {
-            assertThat(expectThrows(VersionConflictEngineException.class, future::actionGet), sameInstance(updateException));
+            // The actionGet method unwraps ES exception to rethrow the cause of it
+            assertThat(expectThrows(VersionConflictEngineException.class, future::actionGet), sameInstance(versionConflictException));
         }
         verify(service, times(2)).shouldSkipUpdateForActivate(any(), any());
         verify(service).doUpdate(any(), anyActionListener());

